### PR TITLE
fix: fix vote delegation

### PR DESCRIPTION
**Pull request by EjembiEmmanuel**
@tensojka I've fixed the the issues with vote delegation. Please review to be sure I'm going in the right direction before I add the tests. Thank you!

### CODE REVIEW

**Review by tensojka — COMMENTED** [review]
Is the only thing you did uncomment the code? Does it work? Did you test it? Not sure what to review.

Sorry for not catching this earlier, please request a review from me (assigning me as a reviewer).

**Comment by EjembiEmmanuel** [comment]
The code now compiles successfully, which was not the case initially. The initial issue was due to the absence of `withdraw_delegation` and `delegate_vote` declarations in the `IProposals` trait.

Additionally, there was an error in the code that retrieves the `gov_token_addr`. The incorrect line of code (line 400) was commented out:
`// let gov_token_addr = self.get_governance_token_address();`

This was corrected on line 401 with the following code:
`let gov_token_addr = IGovernanceDispatcher { contract_address: get_contract_address() }
.get_governance_token_address();`

At this point, I have not yet written tests for the code. I just wanted to get your feedback before proceeding with writing the tests.

File: tests/proposals_tests.cairo [hunk 1]
```diff
@@ -162,3 +162,38 @@ fn test_submit_proposal_under_quorum() {
...
+    let mut calldata: Array<(ContractAddress, u128)> = ArrayTrait::new();
+    let addr: felt252 = 0x4;
+    calldata.append((addr.try_into().unwrap(), 50000));
+    dispatcher
+        .delegate_vote(first_address.try_into().unwrap(), calldata, 50000.try_into().unwrap());
+}
+
+#[test]
+#[should_panic(expected: ('incorrect delegate list',))]
+fn test_withdrae_delegation_with_incorrect_calldata() {
```
Comment (tensojka) [inline tests/proposals_tests.cairo:185]: typo

**Review by tensojka — COMMENTED** [review]
Fine so far. There should be more quite a bit tests though

File: tests/proposals_tests.cairo (hunk 1 above)
Comment (EjembiEmmanuel) [reply tests/proposals_tests.cairo:185]: Oh yea, thanks!

**Review by tensojka — CHANGES_REQUESTED** [review]
Still missing [] Delegate, vote and withdraw test

Otherwise LGTM!

**Comment by EjembiEmmanuel** [comment]
> Still missing [] Delegate, vote and withdraw test
> 
> Otherwise LGTM!

Working on it!

**Review by tensojka — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**